### PR TITLE
[PAN-1930] bootnodes not on whitelist - improve errors

### PR DESCRIPTION
**Pull request by macfarla**
## PR description
Improve testing for errors when bootnodes not in nodes-whitelist

## Fixed Issue(s)
fixes #PAN-2116

### CODE REVIEW

File: pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java [hunk 1]
```diff
@@ -69,7 +69,17 @@ public void nodesWhitelistOptionWhichDoesNotIncludeBootnodesMustError() throws E
           ethNetworkConfig, permissioningConfiguration);
       fail("expected exception because ropsten bootnodes are not in node-whitelist");
     } catch (Exception e) {
-      assertThat(e.getMessage().startsWith("Bootnode")).isTrue();
+      assertThat(e.getMessage().startsWith("Bootnode(s) not in nodes-whitelist")).isTrue();
```
Comment (mark-terry) [inline pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java]: Could it be simplified like:
`assertThat(e.getMessage()).startsWith("Bootnode(s) not in nodes-whitelist")`?

File: pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java [hunk 2]
```diff
@@ -69,7 +69,17 @@ public void nodesWhitelistOptionWhichDoesNotIncludeBootnodesMustError() throws E
...
+      assertThat(
```
Comment (mark-terry) [inline pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java]: See previous.

File: pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java [hunk 3]
```diff
@@ -69,7 +69,17 @@ public void nodesWhitelistOptionWhichDoesNotIncludeBootnodesMustError() throws E
...
+              e.getMessage()
+                  .contains(
+                      "enode://6332792c4a00e3e4ee0926ed89e0d27ef985424d97b6a45bf0f23e51f0dcb5e66b875777506458aea7af6f9e4ffb69f43f3778ee73c81ed9d34c51c4b16b0b0f@52.232.243.152:30303"))
+          .isTrue();
+      assertThat(
```
Comment (mark-terry) [inline pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java]: See previous.

Comment (macfarla) [reply pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java]: 👍

File: pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java (hunk 2 above)
Comment (macfarla) [reply pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java]: 👍

File: pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java (hunk 1 above)
Comment (macfarla) [reply pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java]: 👍

File: pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java [hunk 4]
```diff
@@ -69,7 +69,15 @@ public void nodesWhitelistOptionWhichDoesNotIncludeBootnodesMustError() throws E
           ethNetworkConfig, permissioningConfiguration);
       fail("expected exception because ropsten bootnodes are not in node-whitelist");
     } catch (Exception e) {
-      assertThat(e.getMessage().startsWith("Bootnode")).isTrue();
+      assertThat(e.getMessage().startsWith("Bootnodxe(s) not in nodes-whitelist"));
```
[the remark this answers is not in the document]
Comment (mark-terry) [reply pantheon/src/test/java/tech/pegasys/pantheon/util/PermissioningConfigurationValidatorTest.java]: Also, that assertion isn't asserting anything at the moment.  It should be in the form `assertThat(value).startsWith(test)`, ie:
`assertThat(e.getMessage()).startsWith("Bootnode(s) not in nodes-whitelist")`

**Pull request merged**